### ISSUE TITLE
New product changes disappear when clicking variant before publishing

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#2244]
http://g.recordit.co/JGE9FppgXZ.gif

### Expected behavior
I should be able to edit sections of a product, it's variants, and variant options before publishing.

### Actual Behavior
See gif linked above.
When creating a new product
When I fill out the title, subtitle, vendor, and description and then click the variant box, it clears the information i typed.

Throwing server error:
```
Exception in queued task: Error: Could not find element with id a5K5cTMAQjhD92drn to change
    at [object Object]._.extend.changed (packages/ddp-server/livedata_server.js:188:13)
    at [object Object]._.extend.changed (packages/ddp-server/livedata_server.js:403:10)
    at [object Object]._.extend.changed (packages/ddp-server/livedata_server.js:1262:19)
    at Object.Revisions.find.observe.changed (server/publications/collections/product.js:181:11)
    at [object Object].observeChangesCallbacks.changed (packages/minimongo/observe.js:162:28)
    at self.applyChange.changed (packages/minimongo/observe.js:66:44)
    at packages/mongo/observe_multiplex.js:182:30
    at Array.forEach (native)
    at Function._.each._.forEach (packages/underscore.js:139:11)
    at Object.task (packages/mongo/observe_multiplex.js:176:9)
    at [object Object]._.extend._run (packages/meteor.js:807:18)
    at packages/meteor.js:785:14
```

### Steps to Reproduce the Behavior
1. Clone development
2. Login as admin
3. Create new product
4. Type something into the Title, Subtitle, Vendor, and Description fields.
5. Click the variant box.

### Versions
```
Node: 7.8.0
NPM: 4.5.0
Meteor Node: 4.8.2
Meteor NPM: 4.5.0
Reaction CLI: 0.8.1
Reaction: 1.1.1
Reaction branch: brent-revert-permissions-fix
Docker: 17.03.1-ce
```

**Comment by zenweasel** [comment]
I think this is related to #2079

**Comment by spencern** [comment]
It may be, but I have reproduction steps, and that issue does not. Did my repro steps not work for you?

**Comment by abdulsemiu-atanda** [comment]
@spencern yeah. I tried reproducing on `development` but couldn't

**Comment by spencern** [comment]
I'll check again. Thanks @abdulsemiu-atanda

**Comment by spencern** [comment]
I'm closing this as I think it's related to #2079 as @zenweasel said. I can't reproduce it on fresh install of 1.2

**Comment by zenweasel** [comment]
Closed in favor of #2079
